### PR TITLE
Keep Remote Messages in database if they were shown

### PR DESCRIPTION
**Pull request by ayoy**
Task/Issue URL: https://app.asana.com/0/1207619243206445/1207864976004819/f

**Description**:
Update RemoteMessagingStore to keep track of messages that have ever been shown to the user,
also if the user didn't interact with them (by dismissing or clicking any buttons).
Additionally, upgrade RMF debug view with a button to refetch the config and a bit nicer view
of remote messages in the database.

**Steps to test this PR**:
See [this Asana task](https://app.asana.com/0/1207619243206445/1207887765133538/f).

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/Debug.storyboard [hunk 1]
```diff
@@ -264,7 +264,7 @@
                                             <rect key="frame" x="0.0" y="0.0" width="383.5" height="44.5"/>
                                             <autoresizingMask key="autoresizingMask"/>
                                         </tableViewCellContentView>
-                                        <listContentConfiguration key="contentConfiguration" text="Remote Messages" secondaryText=""/>
+                                        <listContentConfiguration key="contentConfiguration" text="Remote Messaging Framework" secondaryText=""/>
```
Comment (ayoy) [inline DuckDuckGo/Debug.storyboard:267]: Believe me or not, but this is the only change I made to the storyboard 😅

File: DuckDuckGo/HomeMessageViewModel.swift [hunk 2]
```diff
@@ -49,11 +49,6 @@ struct HomeMessageViewModel {
         }
     }
     
-    var topText: String? {
-        // actually unused!
```
Comment (ayoy) [inline DuckDuckGo/HomeMessageViewModel.swift:53]: This is amazing. I decided to action it :)

File: DuckDuckGo/HomeMessageViewSectionRenderer.swift [hunk 3]
```diff
@@ -119,7 +119,16 @@ class HomeMessageViewSectionRenderer: NSObject, HomeViewSectionRenderer {
                 params
             }
         case .remoteMessage(let remoteMessage):
+            let onDidAppear = { [weak self] in
+                self?.homePageConfiguration.didAppear(message)
+            }
+
+            // call didAppear here to support marking messages as shown when they appear on the new tab page
+            // as a result of refreshing a config while the user was on a new tab page already.
+            onDidAppear()
```
Comment (ayoy) [inline DuckDuckGo/HomeMessageViewSectionRenderer.swift:128]: It's fishy, but it works. That said, a remote message appearing on user's new tab page is a rare situation (normally they will appear at app startup after being refreshed in background).

**Review by samsymons — APPROVED** [review]
Works great, and the debug menu changes are nice!

**Pull request merged**